File: test/fixtures/php-guzzle/post_with_quotes_anywhere.php
<?php
require 'vendor/autoload.php';

use GuzzleHttp\Client;

$client = new Client();

$response = $client->post('http://localhost:28139', [
    'headers' => [
        'A'            => "''a'",
        'B'            => '"',
        'Cookie'       => 'x=1\'; y=2"',
        'Content-Type' => 'application/x-www-form-urlencoded'
    ],
    'auth' => ["ol'", 'asd"'],
    'body' => 'a=b&c="&d=\''
]);

File: test/fixtures/java-unirest/post_with_quotes_anywhere.java
import kong.unirest.HttpResponse;
import kong.unirest.Unirest;

class Main {
    public static void main(String[] args) {
            HttpResponse<String> httpResponse
               = Unirest.post("http://localhost:28139")
            .basicAuth("ol'", "asd\"")
            .header("A", "''a'")
            .header("B", "\"")
            .header("Cookie", "x=1'; y=2\"")
            .header("Content-Type", "application/x-www-form-urlencoded")
            .body("a=b&c=\"&d='")
            .asString();
    }
}
